monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunManually/LocalManualRunOptions.js

```diff
@@ -28,29 +29,55 @@ const getContents = (props) => {
   const [osType, setOsType] = useState(OS_TYPES.WINDOWS_64);
   const [selectedIp, setSelectedIp] = useState(props.ips[0]);
   const [commands, setCommands] = useState(generateCommands());
+  const [customUsername, setCustomUsername] = useState('');
+  const customUsernameRef = useRef(null);
 
   useEffect(() => {
     setCommands(generateCommands());
-  }, [osType, selectedIp])
+  }, [osType, selectedIp, customUsername])
 
   function setIp(index) {
     setSelectedIp(props.ips[index]);
   }
 
+  function setUsername() {
+    let inputVal = customUsernameRef.current.value.trim();
+    if (inputVal) {  // checks that it's not just whitespaces
+      setCustomUsername(inputVal);
+    }
+    else {
+      setCustomUsername('');
+    }
+  }
+
   function generateCommands() {
     if (osType === OS_TYPES.WINDOWS_64 || osType === OS_TYPES.WINDOWS_32) {
-      return [{type: 'CMD', command: GenerateLocalWindowsCmd(selectedIp, osType)},
-        {type: 'Powershell', command: GenerateLocalWindowsPowershell(selectedIp, osType)}]
+      return [{type: 'CMD', command: GenerateLocalWindowsCmd(selectedIp, osType, customUsername)},
+        {type: 'Powershell', command: GenerateLocalWindowsPowershell(selectedIp, osType, customUsername)}]
```

Comment:
Why not `customUsernameRef.current.value`?

Reply:
Though we're not using ref anymore, I couldn't do this before since the call to `generateCommands()` in `const [commands, setCommands] = useState(generateCommands());` (line 31) was asking for `customUsernameRef.current.value` without the ref being set which was being done later in the return statement